spec/system/admin/reports/sales_tax/sales_tax_totals_by_order_spec.rb

```diff
@@ -466,15 +468,51 @@ def extract_xlsx_rows(file, range)
       end
 
       context "pdf files" do
+        before do
+          distributor.update(name: "DistributorEnterpriseWithTax")
+          tax_category.update(name: "GST Food")
+        end
         it 'downloads the file' do
-          expect(downloaded_filenames.length).to eq(0) # downloads folder should be empty
           select "PDF", from: "report_format"
-          click_on "Go"
-          wait_for_download
-          expect(downloaded_filenames.length).to eq(1) # downloads folder should contain 1 file
+          expect {
+            generate_report
+          }.to change {
+                 downloaded_filenames.length
+               }.from(0).to(1)
           expect(downloaded_filename).to match(/.*\.pdf/)
+          downloaded_file_pdf = PDF::Reader.new(downloaded_filename)
+
+          # Load PDF pages and contents join into one big string
+          downloaded_file_txt = downloaded_file_pdf.pages.map(&:text).join(" ")
+
+          # It helps to know what we're looking at
+          puts "**DEBUG** downloaded_file_txt:\n #{downloaded_file_txt}"
+
+          expect(downloaded_file_txt).to include "DistributorEnterpriseWithTax", "oc1",
+                                                 "ORDER_NUMBER_1", "GST Food"
+          expect(downloaded_file_txt).to include "State", "cfname", "clname", "ABC123",
+                                                 "order1@example.com"
+          # order1, first tax rate line
+          expect(downloaded_file_txt).to match /1.5 %\s+115.0\s+1.73\s+116.73/
+          # order1, second tax rate line
+          expect(downloaded_file_txt).to match /2.5 %\s+115.0\s+2.88\s+117.88/
+          # correct totals with white-space between them
+          expect(downloaded_file_txt).to match /TOTAL\s+115.0\s+4.61\s+119.61/
+          # order2, first tax rate line
+          expect(downloaded_file_txt).to match /1.5 %\s+215.0\s+3.23\s+218.23/
+          # order2, second tax rate line
+          expect(downloaded_file_txt).to match /2.5 %\s+215.0\s+5.38\s+220.38/
+          # correct totals with white-space between them
+          expect(downloaded_file_txt).to match /TOTAL\s+215.0\s+8.61\s+223.61/
         end
       end
     end
   end
 end
+
+private
+
+def generate_report
+  click_on "Go"
+  wait_for_download
+end
```

Comment:
You are not inside a class here. So the `private` modifier isn't helpful. You are also defining that method globally. It's better to move it into the top `describe` block. Then it can't clash with other specs.